### PR TITLE
Add C# examples in `Basis.xml`

### PR DESCRIPTION
**Pull request by paulloz**
Follow up to #86664

### CODE REVIEW

File: doc/classes/Basis.xml [hunk 1]
```diff
@@ -72,10 +72,18 @@
 			<description>
 				Constructs a pure rotation Basis matrix from Euler angles in the specified Euler rotation order. By default, use YXZ order (most common). See the [enum EulerOrder] enum for possible values.
 				[codeblock]
```
Comment (AThousandShips) [inline doc/classes/Basis.xml:74]:
```suggestion
				[codeblocks]
```
Same everywhere

Comment (paulloz) [reply doc/classes/Basis.xml:74]: Oops, my bad. For future reference, is running `doc/tools/doc_status.py` locally a good way to check everything is formatted correctly?

Comment (AThousandShips) [reply doc/classes/Basis.xml:74]: Haven't tried, might give cryptic results like the CI (which doesn't tell you what's wrong in this case for example)

Comment (akien-mga) [reply doc/classes/Basis.xml]: Not `doc_status.py`, but `make_rst.py` itself reports the errors. Provided you have GNU make and python locally, you can run `make rst` in the `doc` folder.

Comment (paulloz) [reply doc/classes/Basis.xml]: Noted, thanks.

**Review by raulsntos — APPROVED** [review]
LGTM. Thanks!

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**